api/src/main/java/org/apache/cloudstack/api/command/admin/vm/MigrateVMCmd.java

```diff
@@ -169,9 +165,9 @@ public void execute() {
 
         try {
             VirtualMachine migratedVm = null;
-            if (getHostId() != null) {
+            if (getStoragePoolId() == null) {
```

Comment:
@ustcweizhou whenever host / storage pool is not specified in the cmd, instead of directly picking a suitable dest host to migrate to, I think it is better to confirm it (from the user) through a API cmd param (something like _chooseHost_  / _selectHost_).

Reply:
@sureshanaparti ok.
what do you think if we regard hostid=-1 as autoselect ?

Reply:
@weizhouapache host id in the cmd is UUID, so i think it is better to take input from another param when host id is null. If host id is not null, ignore auto selection.

Reply:
@sureshanaparti yes, I am currently working on the coding, same as you said.

ps: cloudstack supports projectid=-1, so it is feasible to support hostid=-1 but it might have big impact on other apis.